x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypeConverter.java

```diff
@@ -55,6 +59,28 @@ public class EsqlDataTypeConverter {
 
     public static final DateFormatter HOUR_MINUTE_SECOND = DateFormatter.forPattern("strict_hour_minute_second_fraction");
 
+    public static final Map<DataType, Integer> dataTypeCastingPriority;
+
+    static {
+        Map<DataType, Integer> map = new HashMap<>();
+        map.put(DataTypes.KEYWORD, 1);
+        map.put(DataTypes.TEXT, 2);
```

Comment:
Why are keyword/text available in the list considering we shouldn't cast to them?

Reply:
keyword/text are removed.